src/Common/src/TypeSystem/Common/TypeSystemHelpers.cs

```diff
@@ -217,5 +218,25 @@ public static string GetFullName(this MetadataType metadataType)
             string ns = metadataType.Namespace;
             return ns.Length > 0 ? String.Concat(ns, ".", metadataType.Name) : metadataType.Name;
         }
+
+        public static IEnumerable<MethodDesc> GetAllVirtualMethods(this TypeDesc type)
```

Comment:
I revised all the places that were calling `GetMethods` but were meaning to get all virtual methods. I'm suspicious about `FindMethodOnExactTypeWithMatchingTypicalMethod` in TypeSystemHelpers.cs though.